library/Backends/Michelson/src/Juvix/Backends/Michelson/DSL/InstructionsEff.hs

```diff
@@ -745,12 +822,18 @@ valToBool _ = error "called valToBool on a non Michelson Bool"
 --------------------------------------------------------------------------------
 
 constructPrim ::
-  (Env.Stack m, Env.Count m, Env.Error m) => Types.RawPrimVal -> Types.Type -> m Env.Expanded
+  Env.Reduction m => Types.RawPrimVal -> Types.Type -> m Env.Expanded
 constructPrim prim ty
+  -- this one gets Promoted to be on the real stack
+  -- TODO ∷ determine if this should be a constant, they should all be in
+  -- vstack...
+  -- also once we do this, we can remove the StackOps effect of this function!
   | isNonFunctionPrim prim = do
-    prim <- Env.Expanded <$> primToArgs prim ty
-    consVal prim ty
-    pure prim
+    primToArgs prim ty
+    -- cons to show we have it on the stack
+    consVal Env.Nop ty
```

Comment:
honestly I just find this pattern confusing (having to use `Nop` like this when we're not actually emitting it as an instr), I guess it's a topic for a later refactor though

Reply:
Yeah, I could put an expanded from the `primToArgs`, however since this pushes to the stack, we just have a nop for anything which likes to process arguments (it's already been processed)

Reply:
it's fine for now